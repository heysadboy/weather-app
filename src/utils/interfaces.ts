import { EActionType, EStatusType } from './enums';

export interface IAction {
    type: EActionType | EStatusType,
    payload: IWeather[] | IWeather | IPage | IStatus
}

export interface IWeather {
    id: number,
    temp: number,
    temp_f: number,
    temp_c: number,
    dt_txt: string,
    tm_txt: string
}

export interface IPage {
    start: number,
    end: number
}

export interface IStatus {
    code: EStatusType
    message: string
}