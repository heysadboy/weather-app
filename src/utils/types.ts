import combineReducers from '../reducers';

export type AppState = ReturnType<typeof combineReducers>;