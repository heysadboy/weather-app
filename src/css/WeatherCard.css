#weather-card {
    padding: 20px 5px 20px 5px;
    min-width: 150px;
}

@media screen and (max-width: 768px) {
    #weather-card {
        padding: 0px;
        min-width: 105px;
    }

    #calendar-icon {
        display: none;
    }
}