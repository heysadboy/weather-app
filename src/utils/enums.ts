export enum EActionType {
    fetch_weather = 'FETCH_WEATHER',
    temp_data = 'SET_TEMP_DATA',
    left_page = 'LEFT_PAGE',
    right_page = 'RIGHT_PAGE'
}

export enum EStatusType {
    error = 'ERROR',
    loading = 'LOADING',
    ok = 'OK'
}

export enum ETempType {
    f = "Fahrenheit",
    c = "Celsius"
}