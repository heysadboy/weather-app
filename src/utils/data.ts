import { IPage, IStatus } from '../utils/interfaces';
import { EStatusType } from './enums';


export const INITIAL_PAGE: IPage = { start: 0, end: 2 }

export const INITAIL_STATUS: IStatus = { code: EStatusType.loading, message: "Loading"}

export const APP_ID = '75f972b80e26f14fe6c920aa6a85ad57';
