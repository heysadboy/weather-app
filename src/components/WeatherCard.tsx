import { FC } from "react";
import { IWeather } from "../utils/interfaces";
import { ETempType } from '../utils/enums';
import '../css/WeatherCard.css';

interface IWeatherCardProp {
    temp_item: IWeather,
    tempType: ETempType
}

const WeatherCard: FC<IWeatherCardProp> = ({ temp_item, tempType }) => {

    const getTypeTemp = () => {
        if (tempType === ETempType.f) {
            let conv_temp = temp_item.temp_f + " °F";
            return conv_temp;
        }
        else {
            let conv_temp = temp_item.temp_c + " °C";
            return conv_temp;
        }
    }

    return (
        <div id="weather-card-container" className="column" key={temp_item.id}>
            <div id="weather-card" className="ui centered card">
                <div className="content">
                    <div className="header">{getTypeTemp()}</div>
                    <div className="description">
                        <p><i id="calendar-icon" className="calendar outline icon" />{temp_item.dt_txt}</p>
                    </div>
                </div>
            </div>
        </div>
    );
}

export default WeatherCard;