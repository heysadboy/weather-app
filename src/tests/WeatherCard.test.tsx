import renderer from 'react-test-renderer';
import WeatherCard from '../components/WeatherCard';
import { ETempType } from '../utils/enums';
import { IWeather } from '../utils/interfaces';


test('Test Weather Card', () => {
    const tempType: ETempType = ETempType.c;
    const temp_item: IWeather = {
        id: 1,
        temp: 0,
        temp_f: 0,
        temp_c: 0,
        dt_txt: "test",
        tm_txt: "test"
    }
    const currentDay: string = "test";
    const setCurrentDay = () => { };

    const component = renderer.create(
        <WeatherCard tempType={tempType} temp_item={temp_item} setCurrentDay={setCurrentDay} currentDay={currentDay} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});