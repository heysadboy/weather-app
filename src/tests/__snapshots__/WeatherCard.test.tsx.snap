// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Weather Card 1`] = `
<div
  className="column"
  id="weather-card-container"
>
  <div
    className="ui centered grey card"
    id="weather-card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        0 Â°C
      </div>
      <div
        className="description"
      >
        <p>
          <i
            className="calendar outline icon"
            id="calendar-icon"
          />
          test
        </p>
      </div>
    </div>
  </div>
</div>
`;
